packages/core/src/subscriber/index.ts

```diff
@@ -0,0 +1,182 @@
+import { utils, config } from '../../';
+import { ethers } from 'ethers';
+import schedule from 'node-schedule';
+
+import { ChainSubscribers } from './chainSubscribers';
+
+let subscribers: ChainSubscribers;
+let usingFallbackUrl = false;
+let waitingForResponseAfterCrash = false;
+let successfullAnswersAfterCrash = 0;
+let successfullAnswersAfterTxRegWarn = 0;
+let failedAnswers = 0;
+let intervalWhenCrash: NodeJS.Timeout | undefined = undefined;
+let intervalWhenTxRegWarn: NodeJS.Timeout | undefined = undefined;
+let waitingForResponseAfterTxRegWarn = false;
+
+let provider: ethers.providers.WebSocketProvider;
+let providerFallback: ethers.providers.WebSocketProvider;
+
+export const start = async (): Promise<void> => {
+    provider = new ethers.providers.WebSocketProvider(config.webSocketUrl);
+    providerFallback = new ethers.providers.WebSocketProvider(
+        config.webSocketUrlFallback
+    );
+    subscribers = startChainSubscriber();
+
+    process.on('unhandledRejection', (error: any) => {
```

Comment:
I actually don't know if it still works in case the provider used is a websocket. Did you tried.
I saw this https://github.com/ethers-io/ethers.js/issues/1053
This https://github.com/ethers-io/ethers.js/issues/1053#issuecomment-861059758 seems simple and might do the work.